src/ui/panels/quests.js

```diff
@@ -0,0 +1,232 @@
+                this.refreshStats();
+                break;
+
+            default:
+                console.log("Error: invalid quest tab.");
+        }
+
+        this.currentTab = tab;
+    }
+
+    // TODO: handle other tabs?
+    closeCurrentTab() {
+        switch (this.currentTab) {
+            case CONSTANTS.QUEST_TABS.QUESTS:
+                this.questsTabButton.setAlpha(0.1);
+                this.questsScrollWindow.setVisible(false);
+                this.isQuestTextVisible = false;
+                break;
+
+            case CONSTANTS.QUEST_TABS.STATS:
+                this.statsTabButton.setAlpha(0.1);
+                this.statsScrollWindow.setVisible(false);
+                this.isStatTextVisible = false;
+                break;
+
+            default:
+                console.log("Error: attempting to close invalid quest tab.");
+        }
+    }
+
+    refreshStats() {
+        this.statsScrollWindow.clearObjects();
+        let statTexts = this.dashboard.scene.get(CONSTANTS.SCENES.STATS).getStats();
+
+        for (const statkey in statTexts) {
+            let sceneText = this.statsScrollWindow.add
+                .text(0, 0, statTexts[statkey], {
+                    fill: "white",
+                    fontSize: 11,
+                })
+                .setDepth(3);
+            this.statsScrollWindow.addObject(sceneText);
+        }
+        this.statsScrollWindow.refresh(true);
+        this.statsScrollWindow.setVisible(this.isStatTextVisible);
+    }
+
+    async refreshQuests() {
+        this.questsScrollWindow.clearObjects();
+        characterData.getUnlockedLevels().forEach((level) => {
+            let scene = this.dashboard.scene.get(level);
+            let enemies = characterData.getEnemiesInLevel(level);
+            for (var enemy in enemies) {
+                let enemiesKilled = characterData.getEnemiesKilled(level, enemy);
+
+                for (
+                    var tier = 1;
+                    tier <=
+                    characterData.calcQuestTier(enemiesKilled, scene.questAmounts[enemy]);
+                    tier++
+                ) {
+                    let questAmount = scene.questAmounts[enemy][tier - 1];
+                    let printedAmount =
+                        enemiesKilled > questAmount ? questAmount : enemiesKilled;
+                    // TODO: make the quests text align so it looks better
+                    let questText = this.questsScrollWindow.add
+                        .text(
+                            0,
+                            0,
+                            printedAmount +
+                                "/" +
+                                questAmount +
+                                " " +
+                                prettyPrintCamelCase(enemy) +
+                                "s",
+                            {
+                                fill: enemiesKilled >= questAmount ? "#00ff00" : "yellow",
```

Comment:
Try changing this formatting to RS font and see if it still looks good

Reply:
Did not look good unfortunately